solidity/contracts/bridge/WalletProposalValidator.sol

```diff
@@ -662,6 +677,61 @@ contract WalletProposalValidator {
         return true;
     }
 
+    function validateMovedFundsSweepProposal(
+        MovedFundsSweepProposal calldata proposal
+    ) external view returns (bool) {
+        Wallets.Wallet memory wallet = bridge.wallets(
+            proposal.walletPubKeyHash
+        );
+
+        // Make sure the wallet is in Live or MovingFunds state.
+        require(
+            wallet.state == Wallets.WalletState.Live ||
+                wallet.state == Wallets.WalletState.MovingFunds,
+            "Source wallet is not in Live or MovingFunds state"
+        );
+
+        // Make sure the moved funds sweep request is valid.
+        uint256 sweepRequestKeyUint = uint256(
+            keccak256(
+                abi.encodePacked(
+                    proposal.movingFundsTxHash,
+                    proposal.movingFundsTxOutputIndex
+                )
+            )
+        );
+
+        MovingFunds.MovedFundsSweepRequest memory sweepRequest = bridge
+            .movedFundsSweepRequests(sweepRequestKeyUint);
+
+        require(
+            sweepRequest.state ==
+                MovingFunds.MovedFundsSweepRequestState.Pending,
+            "Sweep request is not in Pending state"
+        );
+
+        require(
+            sweepRequest.walletPubKeyHash == proposal.walletPubKeyHash,
+            "Sweep request does not belong to the wallet"
+        );
+
```

Comment:
I'm wondering whether we need a kind of safety margin that would prevent handling moved funds sweep requests that are close to reaching their timeout. I mean a similar margin as we use for redemptions (`REDEMPTION_REQUEST_TIMEOUT_SAFETY_MARGIN`).

For redemptions, the goal of the safety margin is to prevent the case where the redeemer reports a timeout, gets their TBTC back, and additionally receives BTC from a belated wallet.

For moved funds sweep, a safety margin would prevent the case where the sweep request timeout is notified but the belated wallet would produce an unprovable Bitcoin transaction right after. The wallet would be anyway terminated and slashed upon timeout notification. A terminated wallet cannot be accused of fraud so, it wouldn't incur any negative consequences for producing an unprovable Bitcoin transaction. Moreover, if the sweep timeout would be a result of a software error rather than an explicit malice, such a wallet may try to recover to unlock funds. Having all funds under one UTXO would be very helpful in such a case. 

That said, I think the safety margin is not needed for this function. Can you double-check and post your thoughts?

Reply:
I also think a timeout margin is unnecessary in case of moved funds sweep validation.
When a wallet performs a moved funds sweep transaction, it sends the funds to itself.
Therefore It shouldn't be a problem - the funds will end up in the wallet address, but as a single UTXO.
As you mentioned, the wallet will be in the `Terminated` state, so it won't be accused of a fraud even though it won't be able to submit the transaction proof.